jax/experimental/mosaic/gpu/launch_context.py

```diff
@@ -337,10 +340,38 @@ def init_tma_desc(host_ptr):
         )
         # TODO(apaszke): Better verification (e.g. slice is non-zero)
         # TODO(apaszke): We always know strides statically.
+        # types based on CUtensorMapDataType
+        # https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TYPES.html#group__CUDA__TYPES_1g42bfc19a0751d7183ee94faf9d9e779d
+        if isinstance(ref_ty.element_type, ir.IntegerType):
+          if reduction_op is not None:
+            raise ValueError(
+                f"TMA with reduction_op={reduction_op} is not supported with Integers"
+            )
+          bitwidth = utils.bitwidth_impl(ref_ty.element_type)
+          if bitwidth == 8:
+            tma_dtype = 0 # uint8
+          elif bitwidth == 16:
+            tma_dtype = 1 # uint16
+          elif bitwidth == 32:
+            tma_dtype = 2 # uint32
+          elif bitwidth == 64:
+            tma_dtype = 4 # uint64
+          elif bitwidth == 4:
+            # NOTE: this dtype diverges from the supported types in cuda and is used
+            # to support int4
+            tma_dtype = 16
+        elif ir.F16Type.isinstance(ref_ty.element_type):
+          tma_dtype = 6
+        elif ir.F32Type.isinstance(ref_ty.element_type):
+          tma_dtype = 7
+        elif ir.BF16Type.isinstance(ref_ty.element_type):
+          tma_dtype = 9
```

Comment:
Could we just assign them our own numbering that is separate from CUDA enums? I'm afraid that in the future TMA will be able to handle more types and we'll end up with 16 being used in the CUDA enum. If we just say this is our own convention it all becomes clearer

Reply:
I think we should also add a check on `jaxlib.__version__` that says that if the version is less than 0.5.4 then we'll still pass in the bitwidth. Otherwise we'll use the enum you have here. If we don't do this, if someone installs an old jaxlib release (which JAX does support), we will end up calling the old runtime functions with the enum values in place of bitwidths.